src/smoothscroll.js

```diff
@@ -0,0 +1,279 @@
+(function(WIN, DOC, undefined) {
```

Comment:
nit: name these `w` and `d`. What happens if someone requires this on the node side while doing server-rendering?